src/itemTypes/webmappingapplication.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ | Copyright 2018 Esri
+ |
+ | Licensed under the Apache License, Version 2.0 (the "License");
+ | you may not use this file except in compliance with the License.
+ | You may obtain a copy of the License at
+ |
+ |    http://www.apache.org/licenses/LICENSE-2.0
+ |
+ | Unless required by applicable law or agreed to in writing, software
+ | distributed under the License is distributed on an "AS IS" BASIS,
+ | WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ | See the License for the specific language governing permissions and
+ | limitations under the License.
+ */
+
+import { IUserRequestOptions } from "@esri/arcgis-rest-auth";
+
+import * as mCommon from "../common";
+import { ITemplate } from "../interfaces";
+
+// -- Exports -------------------------------------------------------------------------------------------------------//
+
+/**
+ * Gets the ids of the dependencies of an AGOL webapp item.
+ *
+ * @param fullItem A webapp item whose dependencies are sought
+ * @param requestOptions Options for requesting information from AGOL
+ * @return A promise that will resolve with list of dependent ids
+ * @protected
+ */
+export function getDependencies (
+  fullItem: ITemplate,
+  requestOptions: IUserRequestOptions
+): Promise<string[]> {
+  return new Promise(resolve => {
+    const dependencies:string[] = [];
+
+    const values = fullItem.data && fullItem.data.values;
+    if (values) {
```

Comment:
we have a `getProp` helper fn in hub.js. It's nice in that you can skip these nested guards... and compose more refined helpers like `getProps`, which them allows you to do things like... (from the Hub's current system of getting item dependencies)

```
function getProps (model, props) {
  return props.reduce((a, p) => {
    let v = getProp(model, p);
    if (v) {
      a.push(v);
    }
    return a;
  }, []);
}
export function processGenericWebApp (model) {
  const props = ['data.webmap', 'data.itemId', 'data.values.webmap'];
  return getProps(model, props);
}

```
Here is where the underlying getProp fn definition is - it's sooo small, it's likely better to copy in vs import another pacakge: 
https://github.com/Esri/hub.js/blob/master/packages/common/src/util.ts#L30

Reply:
Very nice; thanks!